### PR TITLE
Update purl2url support for pypi including download URL(s)

### PR DESCRIPTION
**Pull request by johnmhoran**
Reference: https://github.com/package-url/packageurl-python/issues/143

@keshav-space @TG1999 This covers my initial pypi work.  Cocoapods is next.  I'd planned to include the cocoapods work in this PR along with the pypi work unless you suggest otherwise.

Note that in `purl2url.py` I've added a TODO (with questions/suggestions) in the `build_pypi_download_url()` docstring re whether and if so how we might want to include both `tar.gz` and `.whl` download URLs.  See also my issue comment from 2 weeks ago which raises this issue (https://github.com/package-url/packageurl-python/issues/143#issuecomment-1989761535) -- please let me know what you think.

### CODE REVIEW

File: src/packageurl/contrib/purl2url.py [hunk 1]
```diff
@@ -412,3 +414,38 @@ def get_repo_download_url(purl):
...
+        api_url = f"{base_path}/{name}/json"
+        pypi_metadata = get_response(api_url)
+        download_url = ""
+        for rel in pypi_metadata.get("releases").get(version):
+            if rel.get("packagetype") == "sdist":
+                download_url = rel.get("url")
+                return download_url
+
+
+def get_response(url):
```
Comment (TG1999) [inline src/packageurl/contrib/purl2url.py:446]: I am not sure if we want to do network calls in package-url library @tdruez what's your thoughts on this.

Comment (johnmhoran) [reply src/packageurl/contrib/purl2url.py:446]: @TG1999 @tdruez I am also doing a requests.get for the cocoapods download function I'm currently working on.

Comment (johnmhoran) [reply src/packageurl/contrib/purl2url.py:446]: and I think in addition I need to download and vet the cocoapods JSON in order to ID the homepage URL or similar info in order to locate the cocoapods download repo (if one exists -- for one of the PURL spec examples for cocoapods, if I recall correctly, there is no download available so the generated URL returns a 404).

**Comment by pombredanne** [comment]
@johnmhoran could you resubmit this PR to https://github.com/nexB/fetchcode/ instead? Thanks!

**Comment by johnmhoran** [comment]
@pombredanne I will close this PR after I've tracked the repo URL and download URL changes involved here (and in a local instance) so I can replicate where and as needed.  Earlier this week I opened a fetchcode issue to address the fetchcode part -- https://github.com/nexB/fetchcode/issues/116.

This current purl2url issue originally came from the now-closed [PURL CLI tool issue](https://github.com/nexB/purldb/issues/247#issuecomment-1875899523) in purldb.  I understand that going forward I need to

- add/update the cocoapods and pypi repo-URL-related code in purl2url.py, and
- add/update the cocoapods and pypi download-URL-related code in fetchcode/package.py

Is that an accurate statement of the plan?

**Comment by johnmhoran** [comment]
@pombredanne I've assumed my understanding of the steps (mentioned above) is correct and have deleted my local branch (`143-add-purl2url-package-support`) and created a new branch (`143-add-purl2url-cocoapods-repo-url-support`).

I'll use the new branch for my purl2url work and when ready will open a new PR here, and will add/update the cocoapods and pypi download-URL-related code in fetchcode/package.py.  Closing this PR now.

**Pull request closed without merging**